packages/mobile/src/send/utils.ts

```diff
@@ -0,0 +1,267 @@
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import { FeeType } from 'src/fees/actions'
+import { E164NumberToAddressType, RecipientVerificationStatus } from 'src/identity/reducer'
+import { Recipient, RecipientKind } from 'src/recipients/recipient'
+import {
+  ConfirmationInput,
+  ManuallyValidatedE164NumberToAddress,
+  TransactionData,
+} from 'src/send/reducers'
+import Logger from 'src/utils/Logger'
+
+export const formatDisplayName = (displayName: string) => {
+  if (displayName !== 'Mobile #') {
+    return { displayName, startOfSentenceDisplayName: displayName }
+  }
+
+  return { displayName: 'your contact', startOfSentenceDisplayName: 'Your contract' }
+}
+
+export const getAddressFromPhoneNumber = (
+  e164Number: string,
+  e164NumberToAddress: E164NumberToAddressType,
+  manuallyValidatedE164NumberToAddress: ManuallyValidatedE164NumberToAddress
+): string | null | undefined => {
+  const addresses = e164NumberToAddress[e164Number]
+
+  if (!addresses) {
+    return addresses
+  }
+
+  if (addresses.length === 0) {
+    throw new Error('Phone addresses array should never be empty')
+  }
+
+  if (addresses.length > 1) {
+    Logger.warn('Number mapped to multiple addresses, need to disambiguate')
+    const validatedAddress = manuallyValidatedE164NumberToAddress[e164Number]
+    if (!validatedAddress) {
+      throw new Error(
+        'Multiple addresses but none were manually validated. Should have routed through secure send.'
+      )
+    }
+
+    return validatedAddress
+  }
+  // Normal verified case, return the first address
```

Comment:
Can you add comments like this for all the cases in this function

Reply:
Good suggestion. Done.